### DISCUSSION TITLE
Linter & formatter config

### DISCUSSION THREAD

**Discussion by ttytm** [webui-dev/webui#242] [Ideas]
Would it be okay to add a linter / formatter config and CI for the lib and its examples? 

It would allow running format on save to follow a unifrom code style and take away the mental overhead of this automatable formatting task. A simple github workflow would ensure it's followed. 

For indentation I recommend using tabs, as they have better accessibility. Among the things that are not possible with spaces are that tabs adapt to the indent-width that a developer has configured. Also, the width could be change on the fly - without making a change to the code-base (e.g. when working with splits or in deeply nested code). Tabs are the default for e.g. Go, V, the Linux Kernel, also prettier(probably  themost commonly used formatter for several languages) is about to make tabs the default https://github.com/prettier/prettier/issues/7475.

https://github.com/webui-dev/webui/assets/34311583/887f44d3-1df1-40c5-b8c8-b05e181e095c

For C formatting I would add a `.clang-format` config. With clang it's possible the achieve the optimum for C code base, as it can use white space characters optimally - tabs for indentation spaces for alignment. E.g. the following example formatted by clangd:
![Screenshot_20231007_183351](https://github.com/webui-dev/webui/assets/34311583/64b236f0-f54f-4f81-899f-a5dc59e49a7a)

**Comment by hassandraga** [comment]
Yeah, why not! Sounds reasonable to me.
Let's make this GitHub workflow and use clang-format.

**Reply by ttytm** [reply]
K I'm looking into creating a fitting config that preserves most of the used code style and uses tab indentation.
